libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/notification/NotificationMapper.kt

```diff
@@ -63,16 +66,18 @@ class NotificationContentMapper(
 ) {
     private val timelineEventToNotificationContentMapper = TimelineEventToNotificationContentMapper()
 
-    fun map(notificationEvent: NotificationEvent): NotificationContent =
+    fun map(senderId: UserId, notificationEvent: NotificationEvent): NotificationContent =
         when (notificationEvent) {
-            is NotificationEvent.Timeline -> timelineEventToNotificationContentMapper.map(notificationEvent.event)
+            is NotificationEvent.Timeline -> timelineEventToNotificationContentMapper.map(senderId, notificationEvent.event)
             is NotificationEvent.Invite -> NotificationContent.StateEvent.RoomMemberContent(
                 userId = sessionId.value,
                 membershipState = RoomMembershipState.INVITE,
             )
         }
 }
 
-private fun NotificationItem.timestamp(): Long? {
-    return (this.event as? NotificationEvent.Timeline)?.event?.timestamp()?.toLong()
+private fun NotificationItem.timestamp(clock: SystemClock): Long {
+    // FIXME we can't get the timestamp from the notification item anymore, so we need to fake it
```

Comment:
Sadly, we're going to have to fallback to the current device's system time, as we don't have access to the timestamp anymore. I've asked the Rust team to retrieve it for `NotificationEvent.Timeline` items too.

Reply:
@ganfra should we merge it with this change?

Reply:
I'd say yes